### PR TITLE
add prefer-const rule

### LINKED ISSUE #337: Add prefer-const rule
This is one of most useful rules, we should definitely add it.

https://eslint.org/docs/rules/prefer-const

### PR DESCRIPTION
**Pull request by magurotuna**
Closes #337 

This PR adds [prefer-const](https://eslint.org/docs/rules/prefer-const).

I feel like it's ad-hoc, or a little dirty implementation, and has room for being refactored. But all the tests from ESLint are passed except for one case.

### CODE REVIEW

File: src/rules/prefer_const.rs [hunk 1]
```diff
@@ -0,0 +1,854 @@
...
+            return;
+        }
+        id = setInterval(() => {}, 250);
+    }
+    foo();
+  "#,
+      r#"/*exported a*/ let a; function init() { a = foo(); }"#,
+      // TODO(magurotuna): this is ported from ESLint, but I have no idea why this is valid,
+      // so comment it out for now.
+      // r#"/*exported a*/ let a = 1"#,
```
Comment (magurotuna) [inline src/rules/prefer_const.rs:636]: As I commented, I have no idea why this belongs to "valid" tests :confused: 
https://github.com/eslint/eslint/blob/v7.10.0/tests/lib/rules/prefer-const.js#L70

Comment (bartlomieju) [reply src/rules/prefer_const.rs:636]: @magurotuna this is probably some "magical" ESLint directive that makes it valid - we don't have such thing so you can safely remove this test :)

**Comment by bartlomieju** [comment]
@kdy1 your review would be much appreciated

**Review by kdy1 — APPROVED** [review]
LGTM, I forgot about lazy-initialization while writing a comment, but it seems like you handled it correctly.

File: src/rules/prefer_const.rs (hunk 1 above)
Comment (magurotuna) [reply src/rules/prefer_const.rs:636]: oh, thanks, I didn't know that. I'll remove it.

**Comment by magurotuna** [comment]
I've realized there are a couple of typos, I will fix them as well.

**Comment by bartlomieju** [comment]
@magurotuna LGTM as well. Let me know when you think the PR is ready to be landed

**Comment by magurotuna** [comment]
@bartlomieju 
Done it! I think it's ready, PTAL :)

**Review by bartlomieju — APPROVED** [review]
LGTM, thank you @magurotuna, this rule is very useful!

**Pull request merged**